cron/src/main/java/com/foursquare/service/impl/TrendingCategoriesServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+        } else {
+            calendar.set(Calendar.MONTH, calendar.get(Calendar.MONTH) - 1);
+        }
+
+        List<Venue> allMonthlyVenues = venueRepository.findAllByAddedAtGreaterThanEqual(calendar.getTime());
+
+        if (!allMonthlyVenues.isEmpty()) {
+
+            HashMap<Category, Integer> monthlyCategoriesOfVenues = new HashMap<>();
+
+            for (Venue venue : allMonthlyVenues) {
+                for (Category category : venue.getCategories()) {
+                    if (!monthlyCategoriesOfVenues.containsKey(category)) {
+                        monthlyCategoriesOfVenues.put(category, 0);
+                    }
+                }
+            }
+
+            allMonthlyVenues.stream().forEach(venue ->
+                    venue.getCategories().stream().forEach(category ->
+                            monthlyCategoriesOfVenues.put(category, monthlyCategoriesOfVenues.get(category) + 1)));
+
+            List<Map.Entry<Category, Integer>> sortedMonthlyCategoriesByValue = monthlyCategoriesOfVenues.entrySet().stream()
+                    .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                    .collect(Collectors.toList());
+
+            mailing(getTrendingCategoriesFromSortedList(sortedMonthlyCategoriesByValue));
+        }
+    }
+
+    private List<Map.Entry<Category, Integer>> getTrendingCategoriesFromSortedList(List<Map.Entry<Category, Integer>> sortedCategoriesList) {
+        List<Map.Entry<Category, Integer>> trendingCategories = new ArrayList<>();
+        int indexValue = 0;
+        int trendingCategoriesQuantity = 3;
+
+        if (sortedCategoriesList.size() < 3) {
+            trendingCategoriesQuantity = 1;
+        }
+
+        for(int i = 0; i < trendingCategoriesQuantity; i++) {
+
+            Map.Entry<Category, Integer> entryByIndex = sortedCategoriesList.get(indexValue);
+            trendingCategories.add(entryByIndex);
+
+            for (int a = indexValue + 1; a < sortedCategoriesList.size(); a++) {
+                ++indexValue;
+                if (entryByIndex.getValue() == sortedCategoriesList.get(a).getValue()) {
+                    trendingCategories.add(sortedCategoriesList.get(a));
+                } else {
+                    break;
+                }
+            }
+        }
+
+        return trendingCategories;
+    }
+
+    private void mailing(List<Map.Entry<Category, Integer>> trendingCategories) {
+        SimpleMailMessage message = new SimpleMailMessage();
+
+        List<User> users = userRepository.findAll();
+
+        for(User user : users) {
+            message.setTo(user.getEmail());
+            message.setSubject("Trending monthly categories");
+            List<String> messagesToEmail = new ArrayList<>();
+            for (Map.Entry<Category, Integer> category : trendingCategories) {
+                messagesToEmail.add("Category: " + category.getKey().getName() + "; Quantity: " + category.getValue());
```

Comment:
There's no soul in your messages. Where "Hello, dear <username>! Check our trending categories...."?:)